### PR TITLE
Treat Shift+Backspace as Backspace

### PR DESCRIPTION
**Pull request by Aelnor**
This is a fix for #3652.
I decided not to modify the config file since this is a sane default in my opinion.

### CODE REVIEW

**Comment by Aelnor** [comment]
> Looks like you ran the wrong `rustfmt` config?

Yep, my bad, thanks

**Comment by Aelnor** [comment]
> Could you add a changelog entry?

Done. I'm not sure if I should use single quotes or backticks for the codes. At the moment there are quotes.

File: CHANGELOG.md [hunk 1]
```diff
@@ -26,6 +26,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Low resolution window decoration icon on Windows
 - Mouse bindings for additional buttons need to be specified as a number not a string
 - Don't hide cursor on modifier press with `mouse.hide_when_typing` enabled
+- Shift-Backspace now by default sends '^?' (like normal backspace does) instead of '^H'
```
Comment (kchibisov) [inline CHANGELOG.md:29]:
```suggestion
- `Shift + Backspace` now sends `^?` instead of `^H`
```

Comment (Aelnor) [reply CHANGELOG.md]: Done, thanks

**Pull request merged**